modules/imap_folders/modules.php

```diff
@@ -122,6 +122,40 @@ public function process() {
     }
 }
 
+/**
+ * @subpackage imap_folders/handler
+ */
+class Hm_Handler_process_accept_special_folders extends Hm_Handler_Module {
+    public function process() {
+        list($success, $form) = $this->process_form(array('imap_server_id', 'imap_service_name'));
+        if ($success) {            
+            $cache = Hm_IMAP_List::get_cache($this->cache, $form['imap_server_id']);
+            $imap = Hm_IMAP_List::connect($form['imap_server_id'], $cache);
+
+            if (!is_object($imap) || $imap->get_state() != 'authenticated') {
+                Hm_Msgs('ERRUnable to connect to the selected IMAP server');
+                return;
+            }
+
+            $specials = $this->user_config->get('special_imap_folders', array());
+            if ($exposed = $imap->get_special_use_mailboxes()) {
+                $specials[$form['imap_server_id']] = array('sent' => $exposed['sent'], 'draft' => '', 'trash' => $exposed['trash'], 'archive' => '');
+            } else if ($form['imap_service_name'] == 'gandi') {
```

Comment:
Would be cleaner to store here: https://github.com/jasonmunro/cypht/blob/master/modules/nux/services.php

Reply:
I thought that as imap folder is in charge of setting special folders it could be better to put the code there.

Reply:
ok, let's see what @jasonmunro thinks.

Reply:
this looks fine to me, thanks!